app/src/main/java/com/example/tasky/feature_agenda/data/mapper/AgendaMapper.kt

```diff
@@ -2,24 +2,36 @@ package com.example.tasky.feature_agenda.data.mapper
 
 import android.os.Build
 import androidx.annotation.RequiresApi
+import com.example.tasky.feature_agenda.data.local.entity.AttendeeEntity
+import com.example.tasky.feature_agenda.data.local.entity.EventEntity
+import com.example.tasky.feature_agenda.data.local.entity.ReminderEntity
+import com.example.tasky.feature_agenda.data.local.entity.TaskEntity
+import com.example.tasky.feature_agenda.data.local.relations.EventWithAttendee
 import com.example.tasky.feature_agenda.data.remote.dto.AttendeesDto
 import com.example.tasky.feature_agenda.data.remote.dto.EventDto
 import com.example.tasky.feature_agenda.data.remote.dto.PhotoDto
 import com.example.tasky.feature_agenda.data.remote.dto.ReminderDto
 import com.example.tasky.feature_agenda.data.remote.dto.TaskDto
+import com.example.tasky.feature_agenda.domain.model.AgendaItem
 import com.example.tasky.feature_agenda.domain.model.Attendee
-import com.example.tasky.feature_agenda.domain.model.Event
 import com.example.tasky.feature_agenda.domain.model.Photo
-import com.example.tasky.feature_agenda.domain.model.Reminder
-import com.example.tasky.feature_agenda.domain.model.Task
+import com.example.tasky.feature_agenda.domain.util.ReminderType
 import java.time.Instant
 import java.time.LocalDateTime
 import java.time.ZoneId
+import java.time.ZonedDateTime
+
 
 @RequiresApi(Build.VERSION_CODES.O)
-fun Long.asUtcTimestampToLocalDateTime(): LocalDateTime {
+fun Long.toZonedDateTime(timeZoneId: ZoneId = ZoneId.systemDefault()): ZonedDateTime {
     val instant = Instant.ofEpochMilli(this)
-    return LocalDateTime.ofInstant(instant, ZoneId.of("UTC"))
+    return ZonedDateTime.ofInstant(instant, timeZoneId)
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+fun ZonedDateTime.toLong(): Long {
+    val instant = this.toInstant()
+    return instant.toEpochMilli()
 }
```

Comment:
Here, I'd recommend a name that clarifies more what the `Long` stands for, such as `ZonedDateTime.toUtcTimestamp()`